zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotChunkReader.java

```diff
@@ -82,7 +89,14 @@ public ByteBuffer nextId() {
       return null;
     }
 
-    return encodeChunkId(chunksView.first());
+    final var nextFilePartIndex = Math.ceilDiv(offset, chunkSize) + 1;
+
+    return encodeChunkId(chunksView.first().toString() + "-" + nextFilePartIndex);
+  }
+
+  @Override
+  public void setChunkSize(final int chunkSize) {
+    this.chunkSize = chunkSize;
```

Comment:
:thought_balloon: This class supports seeking to chunks by their id. When seeking, the id and `chunkSize` is used to derive a new offset. That means that assigned chunk ids lose their meaning whenever `chunkSize` changes.
We need to think about this in more detail. Even if it works right now because we only set the chunk size in one specific case where it's probably fine, leaving it like this could be dangerous when making further changes in the future.

Reply:
Perhaps the best solution would be to change chunk ids. Instead of a part number, just use the offset directly. Similarly, `SnapshotChunk` could have `fileOffset` and `fileLenght` instead of `fileBlockIndex` and `totalFileBlocks`.

Reply:
fixed